### PR TITLE
Issue 8951 create account button too wide

### PR DESCRIPTION
**Pull request by haltmar**
### Description

Wrap the "create account" button  display: inline-flex, so that the button only take up as much width as the button content requires, preventing the button from stretching.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [updated screenshot for existing test] Created unit and/or browser tests which fail without the change (if possible) - N/A
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [N/A] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).

#### Database evolutions

Read the guidelines [here](https://github.com/civiform/civiform/wiki/Database#writing-database-evolutions)

No database changes

#### Durable jobs

Read the docs [here](https://github.com/civiform/civiform/wiki/Database#durable-jobs-for-data-updates)

N/A

#### User visible changes

- [N/A] Followed steps to [internationalize](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#application-strings) any new strings
  - [N/A ] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [ N/A ] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [N/A] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [N/A  ] Made equivalent changes in Thymeleaf for applicant-facing features and changes (or created an issue in the [Northstar epic](https://github.com/orgs/civiform/projects/1/views/94) to track that it's missing from Thymeleaf)
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [N/A] Manually evaluated tab order
- [N/A] Manually tested with a screen reader if the feature is applicant-facing. See [screen reader testing](https://github.com/civiform/civiform/wiki/Testing#screen-reader-testing)

#### New Features

N/A

### Instructions for manual testing

Use CiviForm as a guest (do not be logged in)
Apply to a program and submit it
View the "Application Submitted" page

Expected behavior
"Create an account" button is close to the size of its text content.

### Issue(s) this completes

Fixes 8951

### CODE REVIEW

File: server/app/views/applicant/CreateAccountAlertFragment.html [hunk 1]
```diff
@@ -11,13 +11,15 @@ <h2 class="usa-alert__heading" th:text="#{alert.createAccount}"></h2>
...
     <div class="content-spacing"></div>
 
-    <button
-      type="button"
-      class="usa-button usa-button--inverse"
-      th:text="#{button.createAnAccount}"
-      id="create-account"
-      th:attr="data-redirect-to=${createAccountLink}"
-    ></button>
+    <div style="display: inline-flex">
```
Comment (gwendolyngoetz) [inline server/app/views/applicant/CreateAccountAlertFragment.html:14]: Please don't use inline styles.

Comment (melanie-exygy) [inline server/app/views/applicant/CreateAccountAlertFragment.html:14]: **suggestion (blocking):** I agree with Gwen that we don't want to use inline styles.  Here's an interesting article that explains a few reasons why:  [avoiding inline styles](https://www.thoughtco.com/avoid-inline-styles-for-css-3466846).  
I'd suggest taking a look at the [USWDS layout grid](https://designsystem.digital.gov/utilities/layout-grid/) to see if you could use that.  It's what we've been trying to use on the North Star to stay consistent with how we style the layout.  Then, you'll just be able to use their CSS class directly on the element.

File: server/app/views/applicant/CreateAccountAlertFragment.html [hunk 2]
```diff
@@ -11,13 +11,17 @@ <h2 class="usa-alert__heading" th:text="#{alert.createAccount}"></h2>
...
 
-    <button
-      type="button"
-      class="usa-button usa-button--inverse"
-      th:text="#{button.createAnAccount}"
-      id="create-account"
-      th:attr="data-redirect-to=${createAccountLink}"
-    ></button>
+    <div class="grid-row">
+      <div class="grid-col-auto">
```
Comment (melanie-exygy) [inline server/app/views/applicant/CreateAccountAlertFragment.html:15]: **suggestion:** I think you may be able to remove this `div` and just add the `grid-row` `div`.  Would you mind giving that a try? `grid-col-auto` is the default, if I remember correctly.

**Review by melanie-exygy — APPROVED** [review]

File: server/app/views/applicant/CreateAccountAlertFragment.html (hunk 2 above)
Comment (haltmar) [reply server/app/views/applicant/CreateAccountAlertFragment.html:15]: Both Chatgpt and Gemini said "Putting a button directly in a grid-row might lead to unexpected layout issues, especially when responsiveness comes into play. The button might not align or size correctly across different screen sizes."   and recommends the additional <div class="grid-col-auto"> .    Thus keeping it for now.

File: server/app/views/applicant/CreateAccountAlertFragment.html (hunk 1 above)
Comment (haltmar) [reply server/app/views/applicant/CreateAccountAlertFragment.html]: updated per melanie's suggestion

**Comment by haltmar** [comment]
ignore this PR.   review https://github.com/civiform/civiform/compare/main...haltmar:issue-8951-button-too-wide-new instead.

**Comment by haltmar** [comment]
ignore this PR.  please review https://github.com/civiform/civiform/pull/9162

**Comment by haltmar** [comment]
this PR no longer needed.  replaced by https://github.com/civiform/civiform/pull/9162

**Pull request closed without merging**